src/QsCompiler/Core/Dependencies.fs

```diff
@@ -35,184 +46,161 @@ type BuiltIn = {
-        | Value tId -> tId.Namespace.Value = BuiltIn.Test.Namespace.Value && tId.Name.Value = BuiltIn.Test.Name.Value
+        | Value tId -> tId.Namespace.Value = BuiltIn.Test.FullName.Namespace.Value && tId.Name.Value = BuiltIn.Test.FullName.Name.Value
         | Null -> false
 
 
     // hard dependencies in Microsoft.Quantum.Core
 
     static member Length = {
-        Name = "Length" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New)
+        FullName = {Name = "Length" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Function (TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New))
     }
 
     static member RangeReverse = {
-        Name = "RangeReverse" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty    
+        FullName = {Name = "RangeReverse" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Function (TypeParameters = ImmutableArray.Empty)
     }
 
     static member Attribute = {
-        Name = "Attribute" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty
+        FullName = {Name = "Attribute" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Attribute
     }
 
-    static member EntryPoint = {
-        Name = "EntryPoint" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty
+    static member EntryPoint : BuiltIn = {
+        FullName = {Name = "EntryPoint" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Attribute
     }
 
-    static member Deprecated = {
-        Name = "Deprecated" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty
+    static member Deprecated : BuiltIn = {
+        FullName = {Name = "Deprecated" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Attribute
     }
 
-    static member Test = {
-        Name = "Test" |> NonNullable<string>.New
-        Namespace = BuiltIn.DiagnosticsNamespace
-        TypeParameters = ImmutableArray.Empty
+    static member Test : BuiltIn = {
+        FullName = {Name = "Test" |> NonNullable<string>.New; Namespace = BuiltIn.DiagnosticsNamespace}
+        Kind = Attribute
     }
 
     // hard dependencies in Microsoft.Quantum.Canon
 
     static member NoOp = {
-        Name = "NoOp" |> NonNullable<string>.New
-        Namespace = BuiltIn.CanonNamespace
-        TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New)
+        FullName = {Name = "NoOp" |> NonNullable<string>.New; Namespace = BuiltIn.CanonNamespace}
+        Kind = Operation (TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New), IsSelfAdjoint = false)
```

Comment:
No op isn't self adjoint?

Reply:
From looking at https://github.com/microsoft/QuantumLibraries/blob/master/Standard/src/Canon/NoOp.qs
it looks like it isn't declared as such.

Reply:
It should be...